src/main/java/menion/android/whereyougo/network/activity/OpenLocalActivity.java

```diff
@@ -0,0 +1,322 @@
+                        importTask.cancel(true);
+                        importTask = null;
+                    } else {
+                        importTask = new ImportTask(OpenLocalActivity.this, name, size);
+                        importTask.execute(uri);
+                    }
+                    return true;
+
+                }
+            }, null, null, getString(R.string.start), new CustomDialog.OnClickListener() {
+                @Override
+                public boolean onClick(CustomDialog dialog, View v, int btn) {
+                    Intent intent = new Intent(OpenLocalActivity.this, MainActivity.class);
+                    intent.putExtra("cguid", name);
+                    intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP | Intent.FLAG_ACTIVITY_NEW_TASK);
+                    startActivity(intent);
+                    OpenLocalActivity.this.finish();
+                    return true;
+                }
+            });
+            Button buttonDownload = (Button) findViewById(R.id.button_positive);
+            Button buttonStart = (Button) findViewById(R.id.button_negative);
+            buttonStart.setEnabled(cartridgeFile != null);
+
+        } catch (Exception e) {
+            finish();
+            return;
+        }
+    }
+
+    @Override
+    public void onDestroy() {
+        super.onDestroy();
+
+        if (importTask != null && importTask.getStatus() != Status.FINISHED) {
+            importTask.cancel(true);
+            importTask = null;
+        }
+    }
+
+    class ImportTask extends AsyncTask<Uri, ImportTask.Progress, Boolean> {
+        final ProgressDialog progressDialog;
+        final String name;
+        final Long size;
+
+        private String errorMessage;
+
+        public ImportTask(final Context context, String name, long size) {
+            super();
+
+            this.name = name;
+            this.size = size;
+
+            progressDialog = new ProgressDialog(context);
+            progressDialog.setMessage("");
+            progressDialog.setProgressStyle(ProgressDialog.STYLE_HORIZONTAL);
+            progressDialog.setMax(1);
+            progressDialog.setIndeterminate(true);
+            progressDialog.setCanceledOnTouchOutside(false);
+
+            progressDialog.setCancelable(true);
+            progressDialog.setOnCancelListener(new ProgressDialog.OnCancelListener() {
+
+                @Override
+                public void onCancel(DialogInterface arg0) {
+                    if (importTask != null && importTask.getStatus() != Status.FINISHED) {
+                        importTask.cancel(false);
+                        importTask = null;
+                        Log.i("down", "cancel");
+                        ManagerNotify.toastShortMessage(context, getString(R.string.cancelled));
+                    }
+                }
+            });
+        }
+
+        @Override
+        protected Boolean doInBackground(Uri... uris) {
```

Comment:
>(Uri... uris)
What is this?

Reply:
It's a varargs array parameter. It means the parameter is an array, but you can (not must) write function calls like `doInBackground(1,2,3)` instead of `doInBackground(new int[]{1,2,3})`. It seems to be the variant used in AsyncTask, the super class. I reckon it's only ever called with one parameter, also in DownloadCartridgeTask, where the download of serveral GUIDs was implemented.